docs/CrossCompilationModel.md

```diff
@@ -0,0 +1,76 @@
+
+# Crosss-Compilation Model
+
+## Components
+
+When compiling Swift code, the compiler will consult three different sources of
+inputs outside of the user provided code.
+
+1. Compiler Resources
+2. System (C/C++) Headers
+3. Platform Swift Modules (SDK Overlay)
+
+These pieces compose in a particular manner to build a system image to build
+code against.
+
+The compiler resources are content for the compiler that the compiler provides.
```

Comment:
Yeah, but also there is the compatibility required for the standard library. The Swift standard library depends on some of the compiler resources. The reason for calling out the packging here is that at least on Windows, we package the SwiftShims into the SDK to maintain the compatibility with the standard library. But I agree with you that we should make this bit explicit.

Reply:
Yeah... SwiftShims is unfortunate since it combines headers that are definitely compiler-ABI details with headers that are for the stdlib. So for that one, you can flip a coin to choose whether it's compiler-specific or stdlib-specific and you'd be equally right and wrong at the same time.